esp-hal-common/build.rs

```diff
@@ -1,161 +1,142 @@
-fn main() {
-    let esp32 = cfg!(feature = "esp32");
-    let esp32c2 = cfg!(feature = "esp32c2");
-    let esp32c3 = cfg!(feature = "esp32c3");
-    let esp32c6 = cfg!(feature = "esp32c6");
-    let esp32s2 = cfg!(feature = "esp32s2");
-    let esp32s3 = cfg!(feature = "esp32s3");
-
-    // Ensure that exactly one chip has been specified
-    let chip_features = [esp32, esp32c2, esp32c3, esp32c6, esp32s2, esp32s3];
-    match chip_features.iter().filter(|&&f| f).count() {
-        1 => {}
-        n => panic!("Exactly 1 chip must be enabled via its Cargo feature, {n} provided"),
+use std::{fs, path::PathBuf};
+
+use serde::Deserialize;
+
+// Macros taken from:
+// https://github.com/TheDan64/inkwell/blob/36c3b10/src/lib.rs#L81-L110
+
+macro_rules! assert_unique_features {
+    () => {};
+
+    ( $first:tt $(,$rest:tt)* ) => {
+        $(
+            #[cfg(all(feature = $first, feature = $rest))]
+            compile_error!(concat!("Features \"", $first, "\" and \"", $rest, "\" cannot be used together"));
+        )*
+        assert_unique_features!($($rest),*);
+    };
+}
```

Comment:
This macro is also used here: https://github.com/esp-rs/esp-hal/blob/dc0911aa0dcac685475efd37876cd6976d772b30/esp32c3-hal/build.rs#L3-L17

It may be interesting to extract it to a common place to avoid the repetition.

Reply:
I think there's a pretty good chance at this point that we will work on unifying the HALs into a single package, so while this is a good idea I'm not sure it will really matter in practice a couple months down the line. If we end up not unifying things then we will likely create a `eps-hal-build-utils` package or something to that effect, though.